application/models/thread.go

```diff
@@ -147,6 +139,39 @@ func (t *Thread) CreateWithParticipants(postUuid string, user User) error {
 	}
 
 	*t = thread
+	return t.ensureParticipants(post, user.ID)
+}
+
+func (t *Thread) ensureParticipants(post Post, userID int) error {
```

Comment:
Would it be more possible to do something like this?
```go
var tp ThreadParticipant
if err := t.FindByThreadIDAndUserID(t.ID, userID); domain.IsOtherThanNoRows(err) {
    tp.ThreadID = t.ID
    tp.UserID = userID
    tp.Save()
}
```

Reply:
That doesn't appear to get the Post creator included as a participant.

Reply:
OK. I guess that chunk would have to be repeated for the creator.

Reply:
Also, we couldn't do it exactly the way you have it, since `t.FindByThreadIDAndUserID` doesn't return the same kind of error that `IsOtherThanNoRows` is expecting.

Reply:
Good point, `IsOtherThanNoRows` would have to be changed to look for "sql: no rows in result set" within the error message, rather than an exact compare.